components/autofill/src/db/credit_cards.rs

```diff
@@ -95,7 +95,7 @@ pub fn get_all_credit_cards(conn: &Connection) -> Result<Vec<InternalCreditCard>
         let mut stmt = tx.prepare(&sql)?;
         credit_cards = stmt
             .query_map(NO_PARAMS, InternalCreditCard::from_row)?
-            .collect::<Result<Vec<InternalCreditCard>, _>>()?;
+            .collect::<std::result::Result<Vec<InternalCreditCard>, _>>()?;
```

Comment:
I'm lightly surprised that this needed to use an explicit `std::result::Result`, what was the problem with the previous version?

Reply:
Full disclosure, this was a change I made that originated in #3698. I was getting the following build error: ```a value of type `std::result::Result<std::vec::Vec<db::models::credit_card::InternalCreditCard>, error::ErrorKind>` cannot be built from an iterator over elements of type `std::result::Result<db::models::credit_card::InternalCreditCard, rusqlite::Error>` ``` I then prefixed `Result` with `std::result`, made a note to research this, and naturally forgot about it. If there's a more appropriate way to handle this I'm all ears. I'm speaking from the edge of my Rust knowledge at the moment.

Reply:
Thanks. Maybe something not quite working right in the mapping from `rusqlite::Error` to `ErrorKind`. It seems like something we can safely follow up on later.

Reply:
It makes some sense to me - the specific error is `std::result::Result<Vec<InternalCreditCard>, rusqlite::Error>>`, not `std::result::Result<Vec<InternalCreditCard>, ErrorType>>` - there's no automatic conversion there.

Isn't it the same as if we have, say:

```
fn bar() -> std::result::Result<(), rusqlite::Error> {
 Ok(())
}

fn foo() -> Result<()> {
  bar() // doesn't work - wrong E type but not coerced
// but the below does.
  bar()?; // wrong E type but it's coerced here.
  Ok(())
}
```
?

(I really need to study rust's type system)

Reply:
Yeah; I guess I was expecting some implicit conversion via `Into` to happen here, but really I don't have a good reason to expect that.